### ISSUE TITLE
common-grpc: authClient.projectId is not guaranteed to be defined

### ISSUE THREAD

**Issue by ofrobots** [GoogleCloudPlatform/google-cloud-node#2133]
Discovered this while investigating #2125.

[common-grpc:service.js](https://github.com/GoogleCloudPlatform/google-cloud-node/blob/0ff331200c36a590e8f51a3ac0a26f82e33b0ac9/packages/common-grpc/src/service.js#L714):
```js
GrpcService.prototype.getGrpcCredentials_ = function(callback) {
  var self = this;

  this.authClient.getAuthClient(function(err, authClient) {
    if (err) {
      callback(err);
      return;
    }

    var credentials = grpc.credentials.combineChannelCredentials(
      grpc.credentials.createSsl(),
      grpc.credentials.createFromGoogleCredential(authClient)
    );

    if (!self.projectId || self.projectId === '{{projectId}}') {
      self.projectId = authClient.projectId;                  // <<<<<<<<<
    }
```

`authClient.projectId` is not guaranteed to be defined. Unless I am mistaken, only JWTClient would have this property. This means that auto-detection of credentials is not working with gRPC based client libraries unless the user provides the projectId manually?

**Comment by stephenplusplus** [comment]
I actually discovered the same and addressed it as part of #2100 by getting the project ID from `self.authClient.projectId`, the `google-auto-auth` instance property, set here: https://github.com/stephenplusplus/google-auto-auth/blob/2a03c3c9894b44715cae9e67784042d18daceda9/index.js#L87. It seems not all `google-auth-library` auth client instances have the `projectId` as a property, but it will return the `projectId` as a third argument to the `getApplicationDefault` callback. That's the one `google-auto-auth` sets on *its* instance as `projectId`.

> This means that auto-detection of credentials is not working with gRPC based client libraries unless the user provides the projectId manually?

This is true for our APIs that use gax APIs*, but not ones that use gRPC directly through `common.GrpcService`. This is more a question of environment rather than transport mechanism. If `google-auth-library` (not `google-auto-auth`) is working as expected, the callback to `getApplicationDefault()` will return a projectId if one was detected.

\* as we convert, this will be addressed (see a solution in #2072).

**Comment by lukesneeringer** [comment]
It looks like @stephenplusplus is already on this. I was not very clear on the priority based on the description; feel free to change if you think I got it wrong.

**Comment by stephenplusplus** [comment]
Fixed by #2100.
